src/Microsoft.AspNet.Mvc.ModelBinding/Binders/KeyValuePairModelBinder.cs

```diff
@@ -40,7 +40,6 @@ public sealed class KeyValuePairModelBinder<TKey, TValue> : IModelBinder
             {
                 var untypedModel = propertyBindingContext.Model;
                 var model = ModelBindingHelper.CastOrDefault<TModel>(untypedModel);
-                parentBindingContext.ValidationNode.ChildNodes.Add(propertyBindingContext.ValidationNode);
```

Comment:
well, we depend on null check to decide whether to validate or not. In the case of KeyValuePairMB, it will only create the object if it can bind _both_ the key and the value, which means we do not validate if there was a key and there was no value that could be fetched or vice-versa. This is a change in behavior which seems that it is not solvable with the current approach ( not sure even if we should try to solve this).